From 3cb4184d7d31f2467e9c5008089f7b3eebfe1c87 Mon Sep 17 00:00:00 2001
From: glennrp <glennrp>
Date: Fri, 13 Jul 2007 13:35:37 +0000
Subject: [PATCH 01/11] handover to glennrp

---
 README | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/README b/README
index a8e67ca..fb2a31e 100644
--- a/README
+++ b/README
@@ -3,6 +3,9 @@ libmng 1.0.10
 
 Added provisional ANG and anIM support, and made some minor bugfixes.
 
+Gerard has handed libmng maintenance over to Glenn Randers-Pehrson
+(glennrp at users.sourceforge.net)
+
 libmng 1.0.9
 ------------
 
-- 
2.39.2

