From 725563104d4a1b01726b91d5af60ae93147daf16 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 28 Nov 2021 23:41:00 +0700
Subject: [PATCH 1/2] Fix build with old Clang

---
 hwy/targets.cc | 13 +++++++++++++
 hwy/timer.cc   | 13 +++++++++++++
 2 files changed, 26 insertions(+)

diff --git a/hwy/targets.cc b/hwy/targets.cc
index e68f754..195504c 100644
--- a/hwy/targets.cc
+++ b/hwy/targets.cc
@@ -74,7 +74,20 @@ HWY_INLINE void Cpuid(const uint32_t level, const uint32_t count,
   uint32_t b;
   uint32_t c;
   uint32_t d;
+#if HWY_COMPILER_CLANG && !defined __cpuid_count
+#if __i386__
+  __asm("cpuid" : "=a"(a), "=b" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count))
+#else
+  __asm("  xchgq  %%rbx,%q1\n" \
+        "  cpuid\n" \
+        "  xchgq  %%rbx,%q1" \
+      : "=a"(a), "=r" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count));
+#endif
+#else
   __cpuid_count(level, count, a, b, c, d);
+#endif
   abcd[0] = a;
   abcd[1] = b;
   abcd[2] = c;
diff --git a/hwy/timer.cc b/hwy/timer.cc
index 28b5892..dd25600 100644
--- a/hwy/timer.cc
+++ b/hwy/timer.cc
@@ -79,7 +79,20 @@ void Cpuid(const uint32_t level, const uint32_t count,
   uint32_t b;
   uint32_t c;
   uint32_t d;
+#if HWY_COMPILER_CLANG && !defined __cpuid_count
+#if __i386__
+  __asm("cpuid" : "=a"(a), "=b" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count))
+#else
+  __asm("  xchgq  %%rbx,%q1\n" \
+        "  cpuid\n" \
+        "  xchgq  %%rbx,%q1" \
+      : "=a"(a), "=r" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count));
+#endif
+#else
   __cpuid_count(level, count, a, b, c, d);
+#endif
   abcd[0] = a;
   abcd[1] = b;
   abcd[2] = c;
-- 
2.42.0

