From 4bfe489baac903216a244d28671237236d78beed Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 28 Nov 2021 23:41:00 +0700
Subject: [PATCH] Fix build with old Clang

---
 hwy/x86_cpuid.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/hwy/x86_cpuid.h b/hwy/x86_cpuid.h
index 60bd51a0..94a4cda4 100644
--- a/hwy/x86_cpuid.h
+++ b/hwy/x86_cpuid.h
@@ -48,7 +48,20 @@ static inline void Cpuid(const uint32_t level, const uint32_t count,
   uint32_t b;
   uint32_t c;
   uint32_t d;
+#if HWY_COMPILER_CLANG && !defined __cpuid_count
+#if __i386__
+  __asm("cpuid" : "=a"(a), "=b" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count))
+#else
+  __asm("  xchgq  %%rbx,%q1\n" \
+        "  cpuid\n" \
+        "  xchgq  %%rbx,%q1" \
+      : "=a"(a), "=r" (b), "=c"(c), "=d"(d) \
+      : "0"(level), "2"(count));
+#endif
+#else
   __cpuid_count(level, count, a, b, c, d);
+#endif
   abcd[0] = a;
   abcd[1] = b;
   abcd[2] = c;
-- 
2.51.0

