From 510f323edd39d115e41da6fe3ef2afd4ab1ddde8 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Wed, 10 Jul 2024 22:53:32 +0700
Subject: [PATCH 2/2] Fix ppc64 build

---
 hwy/targets.cc | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/hwy/targets.cc b/hwy/targets.cc
index 1e109e7d..c018376e 100644
--- a/hwy/targets.cc
+++ b/hwy/targets.cc
@@ -35,7 +35,22 @@
     HWY_OS_LINUX
 // sys/auxv.h does not always include asm/hwcap.h, or define HWCAP*, hence we
 // still include this directly. See #1199.
-#ifndef TOOLCHAIN_MISS_ASM_HWCAP_H
+#ifndef HWY_HAVE_HWCAP  // allow override
+#ifdef TOOLCHAIN_MISS_ASM_HWCAP_H
+#define HWY_HAVE_HWCAP 0  // CMake failed to find the header
+#elif defined(__has_include)  // note: wrapper macro fails on Clang ~17
+// clang-format off
+#if __has_include(<asm/hwcap.h>)
+// clang-format on
+#define HWY_HAVE_HWCAP 1  // header present
+#else
+#define HWY_HAVE_HWCAP 0  // header not present
+#endif                    // __has_include
+#else                     // compiler lacks __has_include
+#define HWY_HAVE_HWCAP 1  // old defaults
+#endif
+#endif  // HWY_HAVE_HWCAP
+#if HWY_HAVE_HWCAP
 #include <asm/hwcap.h>
 #endif
 #if HWY_HAVE_AUXV
-- 
2.45.2

