From ec3e92627939ac69bb2b1a09dcaf3f4fa54e9f87 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 24 Jul 2023 04:06:45 +0700
Subject: [PATCH 3/3] Fix build with ARM64 MSVC

---
 hwy/base.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hwy/base.h b/hwy/base.h
index a5109ba..dd8e457 100644
--- a/hwy/base.h
+++ b/hwy/base.h
@@ -331,7 +331,7 @@ static constexpr HWY_MAYBE_UNUSED size_t kMaxVectorSize = 16;
 
 // float16_t load/store/conversion intrinsics are always supported on Armv8 and
 // VFPv4. On Armv7 Clang requires __ARM_FP & 2; GCC requires -mfp16-format=ieee.
-#if HWY_ARCH_ARM_A64 ||                                            \
+#if (HWY_ARCH_ARM_A64 && !HWY_COMPILER_MSVC) ||                    \
     (HWY_COMPILER_CLANG && defined(__ARM_FP) && (__ARM_FP & 2)) || \
     HWY_COMPILER_GCC_ACTUAL && defined(__ARM_FP16_FORMAT_IEEE)
 #define HWY_NEON_HAVE_FLOAT16C 1
-- 
2.39.2

