diff -Nur aom-v2.0.0.orig/aom/aom_codec.h aom-v2.0.0/aom/aom_codec.h
--- aom-v2.0.0.orig/aom/aom_codec.h	2020-06-24 21:14:38.000000000 +0700
+++ aom-v2.0.0/aom/aom_codec.h	2020-06-28 20:43:18.127643020 +0700
@@ -78,7 +78,7 @@
 
 /*!\brief Decorator indicating that given struct/union/enum is packed */
 #ifndef ATTRIBUTE_PACKED
-#if defined(__GNUC__) && __GNUC__
+#if defined(__GNUC__) && ((__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 4))
 #define ATTRIBUTE_PACKED __attribute__((packed))
 #elif defined(_MSC_VER)
 #define ATTRIBUTE_PACKED
diff -Nur aom-v2.0.0.orig/av1/encoder/ratectrl.h aom-v2.0.0/av1/encoder/ratectrl.h
--- aom-v2.0.0.orig/av1/encoder/ratectrl.h	2020-06-24 21:14:39.000000000 +0700
+++ aom-v2.0.0/av1/encoder/ratectrl.h	2020-06-27 02:44:52.920384028 +0700
@@ -171,7 +171,7 @@
   int q_1_frame;
   int q_2_frame;
 
-  float_t arf_boost_factor;
+  float arf_boost_factor;
   // Q index used for ALT frame
   int arf_q;
   int active_worst_quality;
