From 5f5958ff76b636b21ab97426fb4779ccd70a04bd Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 28 Nov 2021 21:00:39 +0700
Subject: [PATCH 2/2] Fix build with MSVC

---
 common/args_helper.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/args_helper.c b/common/args_helper.c
index 220186833..c7e008812 100644
--- a/common/args_helper.c
+++ b/common/args_helper.c
@@ -8,6 +8,7 @@
  * Media Patent License 1.0 was not distributed with this source code in the
  * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
  */
+#include "aom_ports/msvc.h"
 #include "common/args_helper.h"
 
 #include <assert.h>
-- 
2.30.2

