From 9913e8dd5e5f4e988121672e87cbd780fd32fa53 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 9 Feb 2026 02:05:09 +0700
Subject: [PATCH] Fix build with MinGW GCC 3.4.2

---
 c/include/brotli/port.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/c/include/brotli/port.h b/c/include/brotli/port.h
index 11cefd7..3e7cf89 100644
--- a/c/include/brotli/port.h
+++ b/c/include/brotli/port.h
@@ -257,7 +257,7 @@
 #if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) &&   \
     !defined(__STDC_NO_VLA__) && !defined(__cplusplus) &&           \
     !defined(__PGI) && !defined(__PGIC__) && !defined(__TINYC__) && \
-    !defined(__clang__)
+    !defined(__clang__) && 0
 #define BROTLI_ARRAY_PARAM(name) (name)
 #else
 #define BROTLI_ARRAY_PARAM(name)
-- 
2.52.0

