diff --git a/include/freetype/internal/ftcalc.h b/include/freetype/internal/ftcalc.h
index d9aea2360..e9697fcd2 100644
--- a/include/freetype/internal/ftcalc.h
+++ b/include/freetype/internal/ftcalc.h
@@ -472,7 +472,7 @@ FT_BEGIN_HEADER
 #endif /* !FT_CONFIG_OPTION_NO_ASSEMBLER */
 
 
-#ifndef FT_MSB
+#if !defined (FT_MSB) || defined (FT_MSB_USE_C_IMPL)
 
   FT_BASE( FT_Int )
   FT_MSB( FT_UInt32  z );
diff --git a/include/ft2build.h b/include/ft2build.h
index 58491ceea..abd1ccf5c 100644
--- a/include/ft2build.h
+++ b/include/ft2build.h
@@ -34,6 +34,10 @@
 #ifndef FT2BUILD_H_
 #define FT2BUILD_H_
 
+#if defined (FT_PREFIX)
+#include "ftprefix.h"
+#endif
+
 #include <freetype/config/ftheader.h>
 
 #endif /* FT2BUILD_H_ */
diff --git a/src/base/ftcalc.c b/src/base/ftcalc.c
index c5bc7e3b1..8c20fc270 100644
--- a/src/base/ftcalc.c
+++ b/src/base/ftcalc.c
@@ -107,7 +107,7 @@
     return a & ~0xFFFFL;
   }
 
-#ifndef FT_MSB
+#if !defined (FT_MSB) || defined (FT_MSB_USE_C_IMPL)
 
   FT_BASE_DEF( FT_Int )
   FT_MSB( FT_UInt32 z )
