From dbed22797d6759fb48a1735ad3eb55be7a560ef8 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 24 Apr 2022 19:09:08 +0700
Subject: [PATCH 2/2] Fix build with MinGW 3.4.2

---
 src/libjasper/base/jas_malloc.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/libjasper/base/jas_malloc.c b/src/libjasper/base/jas_malloc.c
index 9b54000..badebf5 100644
--- a/src/libjasper/base/jas_malloc.c
+++ b/src/libjasper/base/jas_malloc.c
@@ -94,7 +94,6 @@
 #	include <unistd.h>
 #elif defined(_WIN32)
 #	include <windows.h>
-#	include <sysinfoapi.h>
 #endif
 
 /******************************************************************************\
@@ -662,7 +661,14 @@ size_t jas_get_total_mem_size()
 	https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getphysicallyinstalledsystemmemory
 	*/
 	ULONGLONG size;
-	if (!GetPhysicallyInstalledSystemMemory(&size)) {
+	HMODULE hKernel32 = LoadLibraryA("kernel32.dll");
+	typedef BOOL(WINAPI * GetPhysicallyInstalledSystemMemory_t)(PULONGLONG);
+	GetPhysicallyInstalledSystemMemory_t GetPhysicallyInstalledSystemMemory_f =
+	  (GetPhysicallyInstalledSystemMemory_t)(GetProcAddress(hKernel32, "GetPhysicallyInstalledSystemMemory"));
+	if (!GetPhysicallyInstalledSystemMemory_f) {
+		return 0;
+	}
+	if (!GetPhysicallyInstalledSystemMemory_f(&size)) {
 		return 0;
 	}
 	return 1024 * size;
-- 
2.42.0

