From 67c3d086ca90fde679a8669d7d22e722d2180c0e Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 24 Apr 2022 19:09:08 +0700
Subject: [PATCH 2/2] Fix build with MinGW 3.4.2

---
 src/libjasper/base/jas_malloc.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/libjasper/base/jas_malloc.c b/src/libjasper/base/jas_malloc.c
index e889930..eaae843 100644
--- a/src/libjasper/base/jas_malloc.c
+++ b/src/libjasper/base/jas_malloc.c
@@ -94,7 +94,6 @@
 #	include <unistd.h>
 #elif defined(_WIN32)
 #	include <windows.h>
-#	include <sysinfoapi.h>
 #endif
 
 /******************************************************************************\
@@ -662,7 +661,14 @@ size_t jas_get_total_mem_size()
 	https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getphysicallyinstalledsystemmemory
 	*/
 	ULONGLONG mem_size_in_kb;
-	if (!GetPhysicallyInstalledSystemMemory(&mem_size_in_kb)) {
+	HMODULE hKernel32 = LoadLibraryA("kernel32.dll");
+	typedef BOOL(WINAPI * GetPhysicallyInstalledSystemMemory_t)(PULONGLONG);
+	GetPhysicallyInstalledSystemMemory_t GetPhysicallyInstalledSystemMemory_f =
+	  (GetPhysicallyInstalledSystemMemory_t)(GetProcAddress(hKernel32, "GetPhysicallyInstalledSystemMemory"));
+	if (!GetPhysicallyInstalledSystemMemory_f) {
+		return 0;
+	}
+	if (!GetPhysicallyInstalledSystemMemory_f(&mem_size_in_kb)) {
 		return 0;
 	}
 	return (mem_size_in_kb < SIZE_MAX / JAS_CAST(size_t, 1024)) ?
-- 
2.49.0

