From 91f5be1f4c24fd44c52d32bbaea041af97168da6 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Wed, 24 Nov 2021 02:09:39 +0700
Subject: [PATCH] Fix build with MSVC 2019 with __ANSI__ definition

---
 image/sys/ansi.h | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/image/sys/ansi.h b/image/sys/ansi.h
index 74900c9..4400d91 100644
--- a/image/sys/ansi.h
+++ b/image/sys/ansi.h
@@ -27,6 +27,8 @@
 //*@@@---@@@@******************************************************************
 #pragma once
 
+#include <stdint.h>
+
 //================================
 // bitio functions
 //================================
@@ -44,13 +46,8 @@
 //================================
 #define FORCE_INLINE
 #define CDECL
-#if __LP64__
-#define UINTPTR_T unsigned long long
-#define INTPTR_T long long
-#else
-#define UINTPTR_T unsigned int
-#define INTPTR_T int
-#endif
+#define UINTPTR_T uintptr_t
+#define INTPTR_T intptr_t
 
 
 //================================
-- 
2.33.1

