From 0e9214a65a9cbc72611754e6c36d0c70e8636ee4 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 12 Nov 2023 05:01:43 +0700
Subject: [PATCH] Fix build with MSVC 2013

---
 src/read.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/read.c b/src/read.c
index 3d8ffe2..4964870 100644
--- a/src/read.c
+++ b/src/read.c
@@ -38,6 +38,10 @@ static const size_t xmpContentTypeSize = sizeof(xmpContentType);
 // can't be more than 4 unique tuples right now.
 #define MAX_IPMA_VERSION_AND_FLAGS_SEEN 4
 
+#if defined(_MSC_VER) && (_MSC_VER <= 1800)
+#define snprintf _snprintf
+#endif
+
 // ---------------------------------------------------------------------------
 // AVIF codec type (AV1 or AV2)
 
-- 
2.42.0

