From 19ebab9ed15748b727d952ce43bac0dd3cc5af8a Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sat, 22 Mar 2025 19:29:35 +0700
Subject: [PATCH] Disable rand_s on WIN32

---
 expat/lib/xmlparse.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/expat/lib/xmlparse.c b/expat/lib/xmlparse.c
index 890a9389..088e7865 100644
--- a/expat/lib/xmlparse.c
+++ b/expat/lib/xmlparse.c
@@ -135,7 +135,7 @@
 #  include "random_getrandom.h"
 #endif /* defined(HAVE_GETRANDOM) || defined(HAVE_SYSCALL_GETRANDOM) */
 
-#if defined(_WIN32)
+#if defined(_WIN32) && ! defined(XML_POOR_ENTROPY)
 #  include "random_rand_s.h"
 #endif /* defined(_WIN32) */
 
@@ -1089,7 +1089,7 @@ generate_hash_secret_salt(void) {
   return ENTROPY_DEBUG("arc4random", entropy);
 #else
   /* Try high quality providers first .. */
-#  ifdef _WIN32
+#  if defined(_WIN32) && ! defined(XML_POOR_ENTROPY)
   if (writeRandomBytes_rand_s(&entropy, sizeof(entropy))) {
     return ENTROPY_DEBUG("rand_s", entropy);
   }
-- 
2.47.3

