diff -Nur expat-2.2.9_orig/lib/xmlparse.c expat-2.2.9/lib/xmlparse.c
--- expat-2.2.9_orig/lib/xmlparse.c	2019-09-26 05:03:02.000000000 +0700
+++ expat-2.2.9/lib/xmlparse.c	2019-10-24 00:11:42.224621505 +0700
@@ -734,7 +734,7 @@
 
 #endif /* defined(HAVE_ARC4RANDOM) && ! defined(HAVE_ARC4RANDOM_BUF) */
 
-#ifdef _WIN32
+#if defined(_WIN32) && ! defined(XML_POOR_ENTROPY)
 
 /* Obtain entropy on Windows using the rand_s() function which
  * generates cryptographically secure random numbers.  Internally it
@@ -813,7 +813,7 @@
   return ENTROPY_DEBUG("arc4random", entropy);
 #else
   /* Try high quality providers first .. */
-#  ifdef _WIN32
+#  if defined(_WIN32) && ! defined(XML_POOR_ENTROPY)
   if (writeRandomBytes_rand_s((void *)&entropy, sizeof(entropy))) {
     return ENTROPY_DEBUG("rand_s", entropy);
   }
