From 388f9b67dd4242d8706be30b067c05b266e06452 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 23 Jun 2024 19:58:27 +0700
Subject: [PATCH 3/7] Add missing includes

---
 src/imageformats/pfm.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/imageformats/pfm.cpp b/src/imageformats/pfm.cpp
index a04f9ed..7878699 100644
--- a/src/imageformats/pfm.cpp
+++ b/src/imageformats/pfm.cpp
@@ -12,6 +12,9 @@
 #include "pfm_p.h"
 #include "util_p.h"
 
+#include <algorithm>
+#include <cmath>
+
 #include <QColorSpace>
 #include <QDataStream>
 #include <QFloat16>
-- 
2.47.0

