From 59256e43df004c217d6f8b02eb06ced64c678a93 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 23 Jun 2024 19:58:27 +0700
Subject: [PATCH 3/8] Add missing includes

---
 src/imageformats/packbits_p.h | 2 ++
 src/imageformats/util_p.h     | 3 +++
 2 files changed, 5 insertions(+)

diff --git a/src/imageformats/packbits_p.h b/src/imageformats/packbits_p.h
index 63a1b55..7284868 100644
--- a/src/imageformats/packbits_p.h
+++ b/src/imageformats/packbits_p.h
@@ -8,6 +8,8 @@
 #ifndef PACKBITS_P_H
 #define PACKBITS_P_H
 
+#include <cstring>
+
 #include <QIODevice>
 
 /*!
diff --git a/src/imageformats/util_p.h b/src/imageformats/util_p.h
index 710b6e2..0a81b41 100644
--- a/src/imageformats/util_p.h
+++ b/src/imageformats/util_p.h
@@ -8,6 +8,9 @@
 #ifndef UTIL_P_H
 #define UTIL_P_H
 
+#include <algorithm>
+#include <cmath>
+#include <cstring>
 #include <limits>
 
 #include <QImage>
-- 
2.47.3

