From b4dee838a78bd0561822a020245a1309cc309a8d Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 10 Jun 2024 03:39:12 +0700
Subject: [PATCH 4/8] Add __ANSI__ defie to jxr.cpp

---
 src/imageformats/jxr.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/imageformats/jxr.cpp b/src/imageformats/jxr.cpp
index a01aadf..d7b90be 100644
--- a/src/imageformats/jxr.cpp
+++ b/src/imageformats/jxr.cpp
@@ -31,6 +31,9 @@
 #include <QSharedData>
 #include <QTemporaryDir>
 
+#if !defined (__ANSI__)
+#define __ANSI__
+#endif
 #include <JXRGlue.h>
 #include <cstring>
 
-- 
2.51.0

