From 6e91ada3ca30b672d3cb84a7a23522d85909c34a Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 25 Apr 2022 01:44:31 +0700
Subject: [PATCH 6/8] Add ability to disable JXL encoder and threads

---
 src/imageformats/jxl.cpp | 41 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/src/imageformats/jxl.cpp b/src/imageformats/jxl.cpp
index 42176fb..91a3bfa 100644
--- a/src/imageformats/jxl.cpp
+++ b/src/imageformats/jxl.cpp
@@ -15,8 +15,49 @@
 #include "util_p.h"
 
 #include <jxl/cms.h>
+
+#if !defined (DISABLE_JXL_ENCODER)
 #include <jxl/encode.h>
+#else
+typedef struct JxlEncoderStruct JxlEncoder;
+typedef struct JxlEncoderFrameSettingsStruct JxlEncoderFrameSettings;
+typedef struct JxlEncoderOptionsStruct JxlEncoderOptions;
+typedef enum {JXL_ENC_SUCCESS, JXL_ENC_ERROR, JXL_ENC_NEED_MORE_OUTPUT, JXL_ENC_NOT_SUPPORTED} JxlEncoderStatus;
+#define JxlColorEncodingSetToSRGB(_1, _2)
+#define JxlEncoderCreate(_1) nullptr
+#define JxlEncoderDestroy(_1)
+#define JxlEncoderOptionsCreate(_1, _2) nullptr
+#define JxlEncoderOptionsSetDistance(_1, _2)
+#define JxlEncoderInitBasicInfo(_1)
+#define JxlEncoderOptionsSetLossless(_1, _2)
+#define JxlEncoderSetBasicInfo(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderSetICCProfile(_1, _2, _3) JXL_ENC_ERROR
+#define JxlEncoderSetColorEncoding(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderAddImageFrame(_1, _2, _3, _4) JXL_ENC_ERROR
+#define JxlEncoderCloseInput(_1)
+#define JxlEncoderProcessOutput(_1, _2, _3) JXL_ENC_ERROR
+#define JxlEncoderUseContainer(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderSetCodestreamLevel(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderFrameSettingsCreate(_1, _2) nullptr
+#define JxlEncoderSetFrameDistance(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderSetFrameLossless(_1, _2) JXL_ENC_ERROR
+#define JxlEncoderUseBoxes(_1) JXL_ENC_ERROR
+#define JxlEncoderAddBox(_1, _2, _3, _4, _5) JXL_ENC_ERROR
+#define JxlEncoderCloseBoxes(_1)
+#define JxlEncoderCloseFrames(_1)
+#define JxlEncoderInitExtraChannelInfo(_1, _2)
+#define JxlEncoderSetExtraChannelInfo(_1, _2, _3) JXL_ENC_ERROR
+#define JxlEncoderSetExtraChannelBuffer(_1, _2, _3, _4, _5) JXL_ENC_ERROR
+#endif
+#if !defined (DISABLE_JXL_THREADS)
 #include <jxl/thread_parallel_runner.h>
+#else
+#define JxlThreadParallelRunner nullptr
+#define JxlThreadParallelRunnerDestroy(_1)
+#define JxlThreadParallelRunnerCreate(_1, _2) nullptr
+#define JxlDecoderSetParallelRunner(_1, _2, _3) JXL_DEC_SUCCESS
+#define JxlEncoderSetParallelRunner(_1, _2, _3) JXL_ENC_SUCCESS
+#endif
 
 #include <string.h>
 
-- 
2.47.3

