From f44f4a624396f5c31135f21460db5e3b38a69dbc Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 23 Mar 2025 06:30:54 +0700
Subject: [PATCH 7/8] Fix build with libjxl < 0.9.0

---
 src/imageformats/jxl.cpp | 32 +++++++++++++++++++++++++++++---
 1 file changed, 29 insertions(+), 3 deletions(-)

diff --git a/src/imageformats/jxl.cpp b/src/imageformats/jxl.cpp
index bd7e126..8c1c86f 100644
--- a/src/imageformats/jxl.cpp
+++ b/src/imageformats/jxl.cpp
@@ -14,7 +14,9 @@
 #include "microexif_p.h"
 #include "util_p.h"
 
+#if JPEGXL_NUMERIC_VERSION >= JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
 #include <jxl/cms.h>
+#endif
 
 #if !defined (DISABLE_JXL_ENCODER)
 #include <jxl/encode.h>
@@ -75,6 +77,9 @@ Q_LOGGING_CATEGORY(LOG_JXLPLUGIN, "kf.imageformats.plugins.jxl", QtWarningMsg)
 // Define JXL_HDR_PRESERVATION_DISABLED to disable HDR preservation
 // (HDR images are saved as UINT16).
 // #define JXL_HDR_PRESERVATION_DISABLED
+#if JPEGXL_NUMERIC_VERSION >= JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
+#define JXL_HDR_PRESERVATION_DISABLED
+#endif
 #endif
 
 #ifndef JXL_DECODE_BOXES_DISABLED
@@ -299,6 +304,7 @@ bool QJpegXLHandler::countALLFrames()
     bool is_gray = m_basicinfo.num_color_channels == 1 && m_basicinfo.alpha_bits == 0;
     JxlColorEncoding color_encoding;
     if (m_basicinfo.uses_original_profile == JXL_FALSE && m_basicinfo.have_animation == JXL_FALSE) {
+#if JPEGXL_NUMERIC_VERSION >= JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
         const JxlCmsInterface *jxlcms = JxlGetDefaultCms();
         if (jxlcms) {
             status = JxlDecoderSetCms(m_decoder, *jxlcms);
@@ -308,6 +314,7 @@ bool QJpegXLHandler::countALLFrames()
         } else {
             qCWarning(LOG_JXLPLUGIN, "No JPEG XL CMS Interface");
         }
+#endif
 
         JxlColorEncodingSetToSRGB(&color_encoding, is_gray ? JXL_TRUE : JXL_FALSE);
         JxlDecoderSetPreferredColorProfile(m_decoder, &color_encoding);
@@ -368,17 +375,34 @@ bool QJpegXLHandler::countALLFrames()
         }
     }
 
-    status = JxlDecoderGetColorAsEncodedProfile(m_decoder, JXL_COLOR_PROFILE_TARGET_DATA, &color_encoding);
+    status = JxlDecoderGetColorAsEncodedProfile(m_decoder,
+#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
+                                                &m_input_pixel_format,
+#endif
+                                                JXL_COLOR_PROFILE_TARGET_DATA,
+                                                &color_encoding);
 
     if (status == JXL_DEC_SUCCESS && color_encoding.color_space == JXL_COLOR_SPACE_RGB && color_encoding.white_point == JXL_WHITE_POINT_D65
         && color_encoding.primaries == JXL_PRIMARIES_SRGB && color_encoding.transfer_function == JXL_TRANSFER_FUNCTION_SRGB) {
         m_colorspace = QColorSpace(QColorSpace::SRgb);
     } else {
         size_t icc_size = 0;
-        if (JxlDecoderGetICCProfileSize(m_decoder, JXL_COLOR_PROFILE_TARGET_DATA, &icc_size) == JXL_DEC_SUCCESS) {
+        if (JxlDecoderGetICCProfileSize(m_decoder,
+#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
+                                        &m_input_pixel_format,
+#endif
+                                        JXL_COLOR_PROFILE_TARGET_DATA,
+                                        &icc_size)
+            == JXL_DEC_SUCCESS) {
             if (icc_size > 0) {
                 QByteArray icc_data(icc_size, 0);
-                if (JxlDecoderGetColorAsICCProfile(m_decoder, JXL_COLOR_PROFILE_TARGET_DATA, reinterpret_cast<uint8_t *>(icc_data.data()), icc_data.size())
+                if (JxlDecoderGetColorAsICCProfile(m_decoder,
+#if JPEGXL_NUMERIC_VERSION < JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
+                                                                   &m_input_pixel_format,
+#endif
+                                                                   JXL_COLOR_PROFILE_TARGET_DATA,
+                                                                   reinterpret_cast<uint8_t *>(icc_data.data()),
+                                                                   icc_data.size())
                     == JXL_DEC_SUCCESS) {
                     m_colorspace = QColorSpace::fromIccProfile(icc_data);
 
@@ -1879,6 +1903,7 @@ bool QJpegXLHandler::rewind()
             return false;
         }
 
+#if JPEGXL_NUMERIC_VERSION >= JPEGXL_COMPUTE_NUMERIC_VERSION(0, 9, 0)
         const JxlCmsInterface *jxlcms = JxlGetDefaultCms();
         if (jxlcms) {
             status = JxlDecoderSetCms(m_decoder, *jxlcms);
@@ -1888,6 +1913,7 @@ bool QJpegXLHandler::rewind()
         } else {
             qCWarning(LOG_JXLPLUGIN, "No JPEG XL CMS Interface");
         }
+#endif
 
         bool is_gray = m_basicinfo.num_color_channels == 1 && m_basicinfo.alpha_bits == 0;
         JxlColorEncoding color_encoding;
-- 
2.51.1

