From 3ab066b8472f57538220d49dcf834e1ff568028d Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sat, 19 Dec 2020 20:24:59 +0700
Subject: [PATCH 1/2] Fix MSVC crash on static build

---
 libheif/heif_plugin_registry.cc | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/libheif/heif_plugin_registry.cc b/libheif/heif_plugin_registry.cc
index 15a7cf7..f1bce8e 100644
--- a/libheif/heif_plugin_registry.cc
+++ b/libheif/heif_plugin_registry.cc
@@ -72,9 +72,15 @@ std::set<std::unique_ptr<struct heif_encoder_descriptor>,
          encoder_descriptor_priority_order> s_encoder_descriptors;
 
 
-static class Register_Default_Plugins
+class Register_Default_Plugins
 {
 public:
+  static void ensure_registered()
+  {
+    static Register_Default_Plugins dummy;
+  }
+
+private:
   Register_Default_Plugins()
   {
 #if HAVE_LIBDE265
@@ -101,7 +107,7 @@ public:
     heif::register_decoder(get_decoder_plugin_dav1d());
 #endif
   }
-} dummy;
+};
 
 
 void heif::register_decoder(const heif_decoder_plugin* decoder_plugin)
@@ -119,6 +125,7 @@ const struct heif_decoder_plugin* heif::get_decoder(enum heif_compression_format
   int highest_priority = 0;
   const struct heif_decoder_plugin* best_plugin = nullptr;
 
+  Register_Default_Plugins::ensure_registered();
   for (const auto* plugin : s_decoder_plugins) {
     int priority = plugin->does_support_format(type);
     if (priority > highest_priority) {
@@ -161,6 +168,7 @@ heif::get_filtered_encoder_descriptors(enum heif_compression_format format,
 {
   std::vector<const struct heif_encoder_descriptor*> filtered_descriptors;
 
+  Register_Default_Plugins::ensure_registered();
   for (const auto& descr : s_encoder_descriptors) {
     const struct heif_encoder_plugin* plugin = descr->plugin;
 
-- 
2.20.1

