From 969f94ba31b13f6190faf288b45f28f60d134621 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 30 Nov 2020 02:13:09 +0700
Subject: [PATCH] Fix build with MSVC 2019

---
 libheif/heif_encoder_aom.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libheif/heif_encoder_aom.cc b/libheif/heif_encoder_aom.cc
index b15854b..ded79de 100644
--- a/libheif/heif_encoder_aom.cc
+++ b/libheif/heif_encoder_aom.cc
@@ -27,6 +27,7 @@
 #include "config.h"
 #endif
 
+#include <algorithm>
 #include <cstring>
 #include <cstdio>
 #include <cassert>
-- 
2.21.0

