From a14fc11165d01f0a3e14cfc0d282e999492ab232 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sat, 19 Dec 2020 20:24:59 +0700
Subject: [PATCH] Fix MSVC crash on static build

---
 libheif/heif_plugin_registry.cc | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/libheif/heif_plugin_registry.cc b/libheif/heif_plugin_registry.cc
index 58b03fd..559cafa 100644
--- a/libheif/heif_plugin_registry.cc
+++ b/libheif/heif_plugin_registry.cc
@@ -66,14 +66,20 @@ std::multiset<std::unique_ptr<struct heif_encoder_descriptor>,
 
 // Note: we cannot move this to 'heif_init' because we have to make sure that this is initialized
 // AFTER the two global std::set above.
-static class Register_Default_Plugins
+class Register_Default_Plugins
 {
 public:
+  static void ensure_registered()
+  {
+    static Register_Default_Plugins dummy;
+  }
+
+private:
   Register_Default_Plugins()
   {
     register_default_plugins();
   }
-} dummy;
+};
 
 
 void heif::register_default_plugins()
@@ -123,6 +129,7 @@ const struct heif_decoder_plugin* heif::get_decoder(enum heif_compression_format
   int highest_priority = 0;
   const struct heif_decoder_plugin* best_plugin = nullptr;
 
+  Register_Default_Plugins::ensure_registered();
   for (const auto* plugin : s_decoder_plugins) {
     int priority = plugin->does_support_format(type);
     if (priority > highest_priority) {
@@ -165,6 +172,7 @@ heif::get_filtered_encoder_descriptors(enum heif_compression_format format,
 {
   std::vector<const struct heif_encoder_descriptor*> filtered_descriptors;
 
+  Register_Default_Plugins::ensure_registered();
   for (const auto& descr : s_encoder_descriptors) {
     const struct heif_encoder_plugin* plugin = descr->plugin;
 
-- 
2.30.2

