From a51e9db9181acf712f47db0c3af34def27123c85 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Thu, 13 Jun 2024 00:04:24 +0700
Subject: [PATCH 1/4] Switch back from libdeflate to zlib

---
 src/lib/OpenEXRCore/compression.c | 126 +++++-------------------------
 1 file changed, 20 insertions(+), 106 deletions(-)

diff --git a/src/lib/OpenEXRCore/compression.c b/src/lib/OpenEXRCore/compression.c
index dc0c973b..eab434af 100644
--- a/src/lib/OpenEXRCore/compression.c
+++ b/src/lib/OpenEXRCore/compression.c
@@ -16,29 +16,9 @@
 
 #include "OpenEXRConfigInternal.h"
 
-#if OPENEXR_USE_INTERNAL_DEFLATE
-#    include "../../../external/deflate/lib/lib_common.h"
-#    include "../../../external/deflate/lib/arm/cpu_features.c"
-#    include "../../../external/deflate/lib/x86/cpu_features.c"
-#    include "../../../external/deflate/lib/utils.c"
-#    include "../../../external/deflate/lib/deflate_compress.c"
-#    undef BITBUF_NBITS
-#    include "../../../external/deflate/lib/deflate_decompress.c"
-#    undef BITBUF_NBITS
-#    include "../../../external/deflate/lib/adler32.c"
-#    include "../../../external/deflate/lib/zlib_compress.c"
-#    include "../../../external/deflate/lib/zlib_decompress.c"
-#else
-#    include <libdeflate.h>
-#endif
+#include <zlib.h>
 #include <string.h>
 
-#if (                                                                          \
-    LIBDEFLATE_VERSION_MAJOR > 1 ||                                            \
-    (LIBDEFLATE_VERSION_MAJOR == 1 && LIBDEFLATE_VERSION_MINOR > 18))
-#    define EXR_USE_CONFIG_DEFLATE_STRUCT 1
-#endif
-
 /* value Aras found to be better trade off of speed vs size */
 #define EXR_DEFAULT_ZLIB_COMPRESS_LEVEL 4
 
@@ -49,7 +29,7 @@ exr_compress_max_buffer_size (size_t in_bytes)
 {
     size_t r, extra;
 
-    r = libdeflate_zlib_compress_bound (NULL, in_bytes);
+    r = compressBound (in_bytes);
     /*
      * lib deflate has a message about needing a 9 byte boundary
      * but is unclear if it actually adds that or not
@@ -95,19 +75,8 @@ exr_compress_buffer (
     size_t              out_bytes_avail,
     size_t*             actual_out)
 {
-    struct libdeflate_compressor* comp;
-
-#ifdef EXR_USE_CONFIG_DEFLATE_STRUCT
-    struct libdeflate_options opt = {
-        .sizeof_options = sizeof (struct libdeflate_options),
-        .malloc_func    = ctxt ? ctxt->alloc_fn : internal_exr_alloc,
-        .free_func      = ctxt ? ctxt->free_fn : internal_exr_free};
-
-#else
-    libdeflate_set_memory_allocator (
-        ctxt ? ctxt->alloc_fn : internal_exr_alloc,
-        ctxt ? ctxt->free_fn : internal_exr_free);
-#endif
+    int    res   = Z_OK;
+    uLongf outsz = out_bytes_avail;
 
     if (level < 0)
     {
@@ -116,25 +85,11 @@ exr_compress_buffer (
         if (level < 0) level = EXR_DEFAULT_ZLIB_COMPRESS_LEVEL;
     }
 
-#ifdef EXR_USE_CONFIG_DEFLATE_STRUCT
-    comp = libdeflate_alloc_compressor_ex (level, &opt);
-#else
-    comp = libdeflate_alloc_compressor (level);
-#endif
-    if (comp)
+    res = compress2 ((Bytef*)out, &outsz, (const Bytef*)in, in_bytes, level);
+    if (res == Z_OK)
     {
-        size_t outsz;
-        outsz =
-            libdeflate_zlib_compress (comp, in, in_bytes, out, out_bytes_avail);
-
-        libdeflate_free_compressor (comp);
-
-        if (outsz != 0)
-        {
-            if (actual_out) *actual_out = outsz;
-            return EXR_ERR_SUCCESS;
-        }
-        return EXR_ERR_OUT_OF_MEMORY;
+        if (actual_out) *actual_out = outsz;
+        return EXR_ERR_SUCCESS;
     }
     return EXR_ERR_OUT_OF_MEMORY;
 }
@@ -150,60 +105,19 @@ exr_uncompress_buffer (
     size_t              out_bytes_avail,
     size_t*             actual_out)
 {
-    struct libdeflate_decompressor* decomp;
-    enum libdeflate_result          res;
-    size_t                          actual_in_bytes;
-#ifdef EXR_USE_CONFIG_DEFLATE_STRUCT
-    struct libdeflate_options opt = {
-        .sizeof_options = sizeof (struct libdeflate_options),
-        .malloc_func    = ctxt ? ctxt->alloc_fn : internal_exr_alloc,
-        .free_func      = ctxt ? ctxt->free_fn : internal_exr_free};
-#endif
-
-//    if (in_bytes == out_bytes_avail)
-//    {
-//        if (actual_out) *actual_out = in_bytes;
-//        if (in != out)
-//            memcpy(out, in, in_bytes);
-//
-//        return EXR_ERR_SUCCESS;
-//    }
-
-#ifdef EXR_USE_CONFIG_DEFLATE_STRUCT
-    decomp = libdeflate_alloc_decompressor_ex (&opt);
-#else
-    libdeflate_set_memory_allocator (
-        ctxt ? ctxt->alloc_fn : internal_exr_alloc,
-        ctxt ? ctxt->free_fn : internal_exr_free);
-    decomp = libdeflate_alloc_decompressor ();
-#endif
-    if (decomp)
+    int    res             = Z_OK;
+    uLongf outsz           = out_bytes_avail;
+    uLongf actual_in_bytes = in_bytes;
+
+    res = uncompress2 ((Bytef*)out, &outsz, (const Bytef*)in, &actual_in_bytes);
+    if (res == Z_OK)
+    {
+        if (actual_out) *actual_out = outsz;
+        if (in_bytes == actual_in_bytes) return EXR_ERR_SUCCESS;
+        /* it's an error to not consume the full buffer, right? */
+    }
+    if (res == Z_OK || res == Z_DATA_ERROR)
     {
-        res = libdeflate_zlib_decompress_ex (
-            decomp,
-            in,
-            in_bytes,
-            out,
-            out_bytes_avail,
-            &actual_in_bytes,
-            actual_out);
-
-        libdeflate_free_decompressor (decomp);
-
-        if (res == LIBDEFLATE_SUCCESS)
-        {
-            if (in_bytes == actual_in_bytes) return EXR_ERR_SUCCESS;
-            /* it's an error to not consume the full buffer, right? */
-        }
-        else if (res == LIBDEFLATE_INSUFFICIENT_SPACE)
-        {
-            return EXR_ERR_OUT_OF_MEMORY;
-        }
-        else if (res == LIBDEFLATE_SHORT_OUTPUT)
-        {
-            /* TODO: is this an error? */
-            return EXR_ERR_SUCCESS;
-        }
         return EXR_ERR_CORRUPT_CHUNK;
     }
     return EXR_ERR_OUT_OF_MEMORY;
-- 
2.47.3

