From 473cbe9f741bd0ec29423929f709834ab7ccef8d Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 20 Oct 2024 06:07:30 +0700
Subject: [PATCH 2/2] Disable std::filesystem

---
 src/lib/OpenEXR/ImfStdIO.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lib/OpenEXR/ImfStdIO.cpp b/src/lib/OpenEXR/ImfStdIO.cpp
index 9d868500..357f8beb 100644
--- a/src/lib/OpenEXR/ImfStdIO.cpp
+++ b/src/lib/OpenEXR/ImfStdIO.cpp
@@ -14,7 +14,7 @@
 #include <ImfMisc.h>
 #include <ImfStdIO.h>
 #include <errno.h>
-#include <filesystem>
+/*#include <filesystem>*/
 
 using namespace std;
 #include "ImfNamespace.h"
@@ -27,14 +27,14 @@ namespace
 inline ifstream*
 make_ifstream (const char* filename)
 {
-    return new ifstream (std::filesystem::path (filename),
+    return new ifstream (/*std::filesystem::path*/ (filename),
                          ios_base::in | ios_base::binary);
 }
 
 inline ofstream*
 make_ofstream (const char* filename)
 {
-    return new ofstream (std::filesystem::path (filename),
+    return new ofstream (/*std::filesystem::path*/ (filename),
                          ios_base::out | ios_base::binary);
 }
 
-- 
2.47.0

