From e26bee8a0c028fafa4682e89b7a636bf1ed85147 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 20 Oct 2024 06:07:30 +0700
Subject: [PATCH 2/3] Disable std::filesystem

---
 src/lib/OpenEXR/ImfStdIO.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/lib/OpenEXR/ImfStdIO.cpp b/src/lib/OpenEXR/ImfStdIO.cpp
index c5c4a26b..48973805 100644
--- a/src/lib/OpenEXR/ImfStdIO.cpp
+++ b/src/lib/OpenEXR/ImfStdIO.cpp
@@ -14,11 +14,13 @@
 #include <ImfMisc.h>
 #include <ImfStdIO.h>
 #include <errno.h>
+/*
 #include <filesystem>
 #if __cplusplus >= 202002L
 #    include <ranges>
 #    include <span>
 #endif
+*/
 
 using namespace std;
 #include "ImfNamespace.h"
@@ -31,6 +33,7 @@ namespace
 inline ifstream*
 make_ifstream (const char* filename)
 {
+/*
 #if __cplusplus >= 202002L
     auto u8view = ranges::views::transform (span{filename, strlen(filename)},
                                             [](char c) -> char8_t { return c; });
@@ -40,11 +43,14 @@ make_ifstream (const char* filename)
     return new ifstream (filesystem::u8path (filename),
                          ios_base::in | ios_base::binary);
 #endif
+*/
+    return new ifstream (filename, ios_base::in | ios_base::binary);
 }
 
 inline ofstream*
 make_ofstream (const char* filename)
 {
+/*
 #if __cplusplus >= 202002L
     auto u8view = ranges::views::transform (span{filename, strlen(filename)},
                                             [](char c) -> char8_t { return c; });
@@ -54,6 +60,8 @@ make_ofstream (const char* filename)
     return new ofstream (filesystem::u8path (filename),
                          ios_base::out | ios_base::binary);
 #endif
+*/
+    return new ofstream (filename, ios_base::out | ios_base::binary);
 }
 
 void
-- 
2.49.0

