From 94e1944d482e5b0cbc709b6ae622a0fbfe8501b4 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sat, 16 Nov 2019 19:25:07 +0700
Subject: [PATCH 3/7] Fix build with MinGW 4.7.2

---
 OpenEXR/IlmImf/ImfMisc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/OpenEXR/IlmImf/ImfMisc.cpp b/OpenEXR/IlmImf/ImfMisc.cpp
index d2c84787..2ba46085 100644
--- a/OpenEXR/IlmImf/ImfMisc.cpp
+++ b/OpenEXR/IlmImf/ImfMisc.cpp
@@ -1389,7 +1389,7 @@ namespace
 // for architectures that require 32-bit alignment for float reading
 //
 
-struct FBytes { uint8_t b[4]; };
+struct FBytes { unsigned char b[4]; };
 union bytesOrFloat {
   FBytes b;
   float f;
-- 
2.39.0

