From 2376cdea50db4b604c123dd3c1fa587269777b31 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Tue, 22 Oct 2024 23:13:27 +0700
Subject: [PATCH 3/5] Avoid std::bad_any_cast exception introduced in macOS
 10.13

---
 src/lib/OpenEXR/ImfMultiPartInputFile.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/lib/OpenEXR/ImfMultiPartInputFile.cpp b/src/lib/OpenEXR/ImfMultiPartInputFile.cpp
index cf1f1a49..eb2021d5 100644
--- a/src/lib/OpenEXR/ImfMultiPartInputFile.cpp
+++ b/src/lib/OpenEXR/ImfMultiPartInputFile.cpp
@@ -169,7 +169,14 @@ MultiPartInputFile::getInputPart (int partNumber)
         _data->parts[partNumber].file = f;
     }
     else
-        f = std::any_cast<file_storage> (_data->parts[partNumber].file);
+    {
+        file_storage* fp = std::any_cast<file_storage> (&(_data->parts[partNumber].file));
+        if (!fp)
+            THROW (
+                IEX_NAMESPACE::LogicExc,
+                "Bad any_cast");
+        f = *fp;
+    }
 
     // TODO: change to by reference / value semantics
     return f.get();
-- 
2.55.0

