From 296df1d74ba089c0ce995b85286845dcb72efbd5 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 6 Dec 2020 19:57:40 +0700
Subject: [PATCH 5/6] Fix build with Ubuntu 16.04

---
 OpenEXR/IlmImf/ImfDwaCompressor.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/OpenEXR/IlmImf/ImfDwaCompressor.cpp b/OpenEXR/IlmImf/ImfDwaCompressor.cpp
index eb0702ea..77930838 100644
--- a/OpenEXR/IlmImf/ImfDwaCompressor.cpp
+++ b/OpenEXR/IlmImf/ImfDwaCompressor.cpp
@@ -159,6 +159,8 @@
 #include <limits>
 
 #include <cstddef>
+#include <stdint.h>
+#include <inttypes.h>
 
 
 // Windows specific addition to prevent the indirect import of the redefined min/max macros
-- 
2.34.0

