diff -Nur exiv2-0.27.0-Source.orig/include/exiv2/version.hpp exiv2-0.27.0-Source/include/exiv2/version.hpp
--- exiv2-0.27.0-Source.orig/include/exiv2/version.hpp	2018-12-20 20:45:16.000000000 +0700
+++ exiv2-0.27.0-Source/include/exiv2/version.hpp	2019-01-04 20:29:49.841887486 +0700
@@ -155,11 +155,13 @@
       The fourth version number is designated a "tweak" an used by Release Candidates
     */
     EXIV2API bool testVersion(int major, int minor, int patch);
+#if 0
     /*!
       @brief dumpLibraryInfo implements the exiv2 option --version --verbose
              used by exiv2 test suite to inspect libraries loaded at run-time
      */
     EXIV2API void dumpLibraryInfo(std::ostream& os,const exv_grep_keys_t& keys);
+#endif
 }                                       // namespace Exiv2
 
 
diff -Nur exiv2-0.27.0-Source.orig/src/version.cpp exiv2-0.27.0-Source/src/version.cpp
--- exiv2-0.27.0-Source.orig/src/version.cpp	2019-01-04 20:29:00.377994037 +0700
+++ exiv2-0.27.0-Source/src/version.cpp	2019-01-04 20:31:10.661761790 +0700
@@ -95,6 +95,7 @@
 
 }                                       // namespace Exiv2
 
+#if 0
 #ifndef lengthof
 #define lengthof(x) sizeof(x)/sizeof(x[0])
 #endif
@@ -557,3 +558,4 @@
 #endif
 
 }
+#endif
