diff -Nur exiv2-0.27.0-Source.orig/include/exiv2/futils.hpp exiv2-0.27.0-Source/include/exiv2/futils.hpp
--- exiv2-0.27.0-Source.orig/include/exiv2/futils.hpp	2019-01-05 02:45:47.701603800 +0700
+++ exiv2-0.27.0-Source/include/exiv2/futils.hpp	2019-01-05 02:46:38.009221000 +0700
@@ -175,8 +175,10 @@
      */
     EXIV2API std::string strError();
 
+#if 0
     //! @brief Return the path of the current process.
     EXIV2API std::string getProcessPath();
+#endif
 
     /*!
       @brief A container for URL components. It also provides the method to parse a
diff -Nur exiv2-0.27.0-Source.orig/src/futils.cpp exiv2-0.27.0-Source/src/futils.cpp
--- exiv2-0.27.0-Source.orig/src/futils.cpp	2019-01-05 02:45:47.710392800 +0700
+++ exiv2-0.27.0-Source/src/futils.cpp	2019-01-05 02:46:38.015080300 +0700
@@ -460,6 +460,7 @@
         return result;
     }
 
+#if 0
     std::string getProcessPath()
     {
         std::string ret("unknown");
@@ -497,4 +498,5 @@
     #endif
         return ret.substr(0, idxLastSeparator);
     }
+#endif
 }                                       // namespace Exiv2
