From 8dad857186a3efc16c3a6ab0ae251dc31562f8a1 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sat, 7 Feb 2026 20:19:17 +0700
Subject: [PATCH 4/4] Fix build with TDM-GCC 10.3

---
 src/crwimage_int.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/crwimage_int.cpp b/src/crwimage_int.cpp
index 9e2c1c6a4..b0f36b0c8 100644
--- a/src/crwimage_int.cpp
+++ b/src/crwimage_int.cpp
@@ -750,7 +750,12 @@ void CrwMap::decode0x180e(const CiffComponent& ciffComponent, const CrwMapping*
   time_t t = v.value_.at(0);
   tm r;
 #ifdef _WIN32
+#ifdef _MSC_VER
   auto tm = localtime_s(&r, &t) ? nullptr : &r;
+#else
+  (void)r;
+  auto tm = std::localtime(&t);
+#endif
 #else
   auto tm = localtime_r(&t, &r);
 #endif
-- 
2.47.3

