From ca0d5a0a6f5cbdbb82f48e1ec6a5853dded418fa Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 22 Jul 2019 04:36:47 +0700
Subject: [PATCH 5/9] Fix launch on Win98

---
 src/makernote_int.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/makernote_int.cpp b/src/makernote_int.cpp
index 8b6e9adc..ecf4b23a 100644
--- a/src/makernote_int.cpp
+++ b/src/makernote_int.cpp
@@ -97,10 +97,14 @@ namespace Exiv2 {
             std::string inifile;
 #if defined(_MSC_VER) || defined(__MINGW__)
             char path[MAX_PATH];
-            if (SUCCEEDED(SHGetFolderPathA(NULL, CSIDL_PROFILE, NULL, 0, path))) {
+            HMODULE hShell32 = ::LoadLibraryA("shell32.dll");
+            typedef HRESULT(WINAPI *SHGetFolderPathA_t)(HWND, int, HANDLE, DWORD, LPSTR);
+            SHGetFolderPathA_t SHGetFolderPathA_f = reinterpret_cast<SHGetFolderPathA_t>(::GetProcAddress(hShell32, "SHGetFolderPathA"));
+            if (SHGetFolderPathA_f && SUCCEEDED(SHGetFolderPathA_f(NULL, CSIDL_PROFILE, NULL, 0, path))) {
                 homedir = std::string(path);
                 inifile = "exiv2.ini"      ;
             }
+            ::FreeLibrary(hShell32);
 #else
             struct passwd* pw = getpwuid(getuid());
             homedir = std::string(pw?pw->pw_dir:"");
-- 
2.39.2

