From 142e1de6fee0332cde4fe1dcc5f7a8ad18ee11e1 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 22 Jul 2019 04:37:35 +0700
Subject: [PATCH 06/10] Use Qt convertStringCharset

---
 src/convert.cpp | 97 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 97 insertions(+)

diff --git a/src/convert.cpp b/src/convert.cpp
index 038d86bf..4ae5a1da 100644
--- a/src/convert.cpp
+++ b/src/convert.cpp
@@ -57,6 +57,19 @@
 # include <errno.h>
 #endif
 
+#if !(defined WIN32 && !defined __CYGWIN__) && !defined EXV_HAVE_ICONV
+# define EXV_HAVE_QT
+# include <QByteArray>
+# include <QString>
+# if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+#  include <QTextCodec>
+# else
+#  include <QSharedPointer>
+#  include <QStringConverter>
+# endif
+# include <QMap>
+#endif
+
 // Adobe XMP Toolkit
 #ifdef EXV_HAVE_XMP_TOOLKIT
 # define TXMP_STRING_TYPE std::string
@@ -78,6 +91,10 @@ namespace {
 #if defined EXV_HAVE_ICONV
     // Convert string charset with iconv.
     bool convertStringCharsetIconv(std::string& str, const char* from, const char* to);
+#endif
+#if defined EXV_HAVE_QT
+    // Convert string charset with QTextCodec/QStringConverter.
+    bool convertStringCharsetQt(std::string& str, const char* from, const char* to);
 #endif
     /*!
       @brief Get the text value of an XmpDatum \em pos.
@@ -1356,6 +1373,8 @@ namespace Exiv2 {
         ret = convertStringCharsetIconv(str, from, to);
 #elif defined WIN32 && !defined __CYGWIN__
         ret = convertStringCharsetWindows(str, from, to);
+#elif defined EXV_HAVE_QT
+        ret = convertStringCharsetQt(str, from, to);
 #else
 # ifndef SUPPRESS_WARNINGS
         EXV_WARNING << "Charset conversion required but no character mapping functionality available.\n";
@@ -1573,6 +1592,84 @@ namespace {
     }
 
 #endif // EXV_HAVE_ICONV
+#if defined EXV_HAVE_QT
+#if(QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+    static QTextCodec* codecForCharset(const char* charset)
+    {
+        if (QTextCodec* codec = QTextCodec::codecForName(charset))
+            return codec;
+
+        QMap<QByteArray, QByteArray> codecsMap;
+        codecsMap["UTF-8"]      = "UTF-8";
+        codecsMap["UCS-2BE"]    = "UTF-16BE";
+        codecsMap["UCS-2LE"]    = "UTF-16LE";
+        codecsMap["ISO-8859-1"] = "ISO-8859-1";
+        codecsMap["ASCII"]      = "ISO-8859-1";
+
+        QMap<QByteArray, QByteArray>::ConstIterator it = codecsMap.find(charset);
+        if (it != codecsMap.constEnd())
+            return QTextCodec::codecForName(it.value());
+
+        return NULL;
+    }
+
+    bool convertStringCharsetQt(std::string& str, const char* from, const char* to)
+    {
+        if (strcmp(from, to) == 0)
+            return true;
+
+        QTextCodec* inCodec = codecForCharset(from);
+        QTextCodec* outCodec = codecForCharset(to);
+        if (!inCodec || !outCodec)
+            return false;
+
+        const QByteArray inArray = QByteArray(str.c_str(), static_cast<int>(str.length()));
+        const QByteArray outArray = outCodec->fromUnicode(inCodec->toUnicode(inArray));
+        str = std::string(outArray.constData(), static_cast<size_t>(outArray.size()));
+        return true;
+    }
+#else
+    template<typename T>
+    static QSharedPointer<T> converterForCharset(const char* charset)
+    {
+        QSharedPointer<T> result(new T(charset));
+        if (result->isValid())
+            return result;
+
+        QMap<QByteArray, QByteArray> codecsMap;
+        codecsMap["UTF-8"]      = "UTF-8";
+        codecsMap["UCS-2BE"]    = "UTF-16BE";
+        codecsMap["UCS-2LE"]    = "UTF-16LE";
+        codecsMap["ISO-8859-1"] = "ISO-8859-1";
+        codecsMap["ASCII"]      = "ISO-8859-1";
+
+        QMap<QByteArray, QByteArray>::ConstIterator it = codecsMap.find(charset);
+        if (it != codecsMap.constEnd())
+            result.reset(new T(it.value()));
+        else
+            result.reset();
+
+        return result;
+    }
+
+    bool convertStringCharsetQt(std::string& str, const char* from, const char* to)
+    {
+        if (strcmp(from, to) == 0)
+            return true;
+
+        QSharedPointer<QStringDecoder> decoder = converterForCharset<QStringDecoder>(from);
+        QSharedPointer<QStringEncoder> encoder = converterForCharset<QStringEncoder>(to);
+        if (!decoder || !encoder)
+            return false;
+
+        const QByteArray inArray = QByteArray(str.c_str(), static_cast<int>(str.length()));
+        const QByteArray outArray = encoder->encode(decoder->decode(inArray));
+        str = std::string(outArray.constData(), static_cast<size_t>(outArray.size()));
+        return true;
+    }
+#endif
+
+#endif // EXV_HAVE_QT
     bool getTextValue(std::string& value, const XmpData::iterator& pos)
     {
         if (pos->typeId() == langAlt) {
-- 
2.34.0

