From ddf0d540cb57b2f5b2dc27af5b042f3ba4f18125 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 22 Jul 2019 04:37:35 +0700
Subject: [PATCH 6/9] Use Qt convertStringCharset

---
 src/convert.cpp | 51 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 51 insertions(+)

diff --git a/src/convert.cpp b/src/convert.cpp
index ef83a23f..1d13f42d 100644
--- a/src/convert.cpp
+++ b/src/convert.cpp
@@ -57,6 +57,14 @@
 # include <errno.h>
 #endif
 
+#if !(defined WIN32 && !defined __CYGWIN__) && !defined EXV_HAVE_ICONV
+# define EXV_HAVE_QT
+# include <QByteArray>
+# include <QString>
+# include <QTextCodec>
+# include <QMap>
+#endif
+
 // Adobe XMP Toolkit
 #ifdef EXV_HAVE_XMP_TOOLKIT
 # define TXMP_STRING_TYPE std::string
@@ -78,6 +86,10 @@ namespace {
 #if defined EXV_HAVE_ICONV
     // Convert string charset with iconv.
     bool convertStringCharsetIconv(std::string& str, const char* from, const char* to);
+#endif
+#if defined EXV_HAVE_QT
+    // Convert string charset with QTextCodec.
+    bool convertStringCharsetQTextCodec(std::string& str, const char* from, const char* to);
 #endif
     /*!
       @brief Get the text value of an XmpDatum \em pos.
@@ -1353,6 +1365,8 @@ namespace Exiv2 {
         ret = convertStringCharsetIconv(str, from, to);
 #elif defined WIN32 && !defined __CYGWIN__
         ret = convertStringCharsetWindows(str, from, to);
+#elif defined EXV_HAVE_QT
+        ret = convertStringCharsetQTextCodec(str, from, to);
 #else
 # ifndef SUPPRESS_WARNINGS
         EXV_WARNING << "Charset conversion required but no character mapping functionality available.\n";
@@ -1570,6 +1584,43 @@ namespace {
     }
 
 #endif // EXV_HAVE_ICONV
+#if defined EXV_HAVE_QT
+    static QTextCodec* codecForCharset(const char* charset)
+    {
+        if (QTextCodec* codec = QTextCodec::codecForName(charset))
+            return codec;
+
+        QMap<QByteArray, QByteArray> codecsMap;
+        codecsMap["UTF-8"]      = "UTF-8";
+        codecsMap["UCS-2BE"]    = "UTF-16BE";
+        codecsMap["UCS-2LE"]    = "UTF-16LE";
+        codecsMap["ISO-8859-1"] = "ISO-8859-1";
+        codecsMap["ASCII"]      = "ISO-8859-1";
+
+        QMap<QByteArray, QByteArray>::ConstIterator it = codecsMap.find(charset);
+        if (it != codecsMap.constEnd())
+            return QTextCodec::codecForName(it.value());
+
+        return NULL;
+    }
+
+    bool convertStringCharsetQTextCodec(std::string& str, const char* from, const char* to)
+    {
+        if (strcmp(from, to) == 0)
+            return true;
+
+        QTextCodec* inCodec = codecForCharset(from);
+        QTextCodec* outCodec = codecForCharset(to);
+        if (!inCodec || !outCodec)
+            return false;
+
+        const QByteArray inArray = QByteArray(str.c_str(), static_cast<int>(str.length()));
+        const QByteArray outArray = outCodec->fromUnicode(inCodec->toUnicode(inArray));
+        str = std::string(outArray.constData(), static_cast<size_t>(outArray.size()));
+        return true;
+    }
+
+#endif // EXV_HAVE_QT
     bool getTextValue(std::string& value, const XmpData::iterator& pos)
     {
         if (pos->typeId() == langAlt) {
-- 
2.20.1

