diff -Nur exiv2-0.27.0-Source.orig/xmpsdk/src/MD5.cpp exiv2-0.27.0-Source/xmpsdk/src/MD5.cpp
--- exiv2-0.27.0-Source.orig/xmpsdk/src/MD5.cpp	2019-01-05 23:54:36.000000000 +0700
+++ exiv2-0.27.0-Source/xmpsdk/src/MD5.cpp	2019-01-05 23:55:30.000000000 +0700
@@ -153,7 +153,7 @@
 void
 MD5Transform(UWORD32 buf[4], UWORD32 const in[16])
 {
-	register UWORD32 a, b, c, d;
+	UWORD32 a, b, c, d;
 
 	a = buf[0];
 	b = buf[1];
