From a735b7a9fd4a8ca631bfccf2e5ec3ce7af8617f2 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 22 Jul 2019 04:38:50 +0700
Subject: [PATCH 07/10] Dont use register keyword

---
 xmpsdk/src/MD5.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xmpsdk/src/MD5.cpp b/xmpsdk/src/MD5.cpp
index f162b728..b040bd2c 100644
--- a/xmpsdk/src/MD5.cpp
+++ b/xmpsdk/src/MD5.cpp
@@ -153,7 +153,7 @@ MD5Final(md5byte digest[16], struct MD5_CTX *ctx)
 void
 MD5Transform(UWORD32 buf[4], UWORD32 const in[16])
 {
-	register UWORD32 a, b, c, d;
+	UWORD32 a, b, c, d;
 
 	a = buf[0];
 	b = buf[1];
-- 
2.34.0

