From ae6e64e394b307553cd4396e07b8da8079014e13 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Mon, 22 Jul 2019 04:41:51 +0700
Subject: [PATCH 9/9] Fix build with Clang 3.3

---
 src/safe_op.hpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/safe_op.hpp b/src/safe_op.hpp
index 116e81fe..ae50caff 100644
--- a/src/safe_op.hpp
+++ b/src/safe_op.hpp
@@ -205,7 +205,9 @@ namespace Safe
         }
 
 #if defined(__GNUC__) || defined(__clang__)
-#if __GNUC__ >= 5 || __clang_major__ >= 3
+#if (defined(__GNUC__) && (__GNUC__ >= 5)) || \
+    (defined(__clang__) && !defined(__apple_build_version__) && ((__clang_major__ > 3) || (__clang_major__ == 3 && __clang_minor__ >= 4))) || \
+    (defined(__clang__) && defined(__apple_build_version__) && (__clang_major__ >= 7))
 
 /*!
  * This macro pastes a specialization of builtin_add_overflow using gcc's &
-- 
2.20.1

