From 03ce28a7533386c29256965350db1547771f800a Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Tue, 29 Dec 2020 17:43:53 +0700
Subject: [PATCH] Fix build with MinGW 3.4.2

---
 libtiff/tif_win32.c | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/libtiff/tif_win32.c b/libtiff/tif_win32.c
index d617a0f4..7ceb0bfd 100644
--- a/libtiff/tif_win32.c
+++ b/libtiff/tif_win32.c
@@ -157,10 +157,27 @@ static uint64_t
 _tiffSizeProc(thandle_t fd)
 {
 	LARGE_INTEGER m;
-	if (GetFileSizeEx(fd,&m))
-		return(m.QuadPart);
+	uint64 result;
+	HMODULE hKernel32 = LoadLibraryA("kernel32.dll");
+	typedef BOOL(WINAPI* GetFileSizeEx_t)(HANDLE, PLARGE_INTEGER);
+	GetFileSizeEx_t GetFileSizeEx_f = (GetFileSizeEx_t)(GetProcAddress(hKernel32, "GetFileSizeEx"));
+	if (GetFileSizeEx_f)
+	{
+		if (GetFileSizeEx_f(fd, &m))
+			result = (uint64)m.QuadPart;
+		else
+			result = 0;
+	}
 	else
-		return(0);
+	{
+		m.LowPart = GetFileSize(fd, (LPDWORD)(&m.HighPart));
+		if (m.LowPart != INVALID_FILE_SIZE)
+			result = (uint64)m.QuadPart;
+		else
+			result = 0;
+	}
+	FreeLibrary(hKernel32);
+	return (result);
 }
 
 static int
-- 
2.37.1

