From 981bd31d5c9eb21a6eff902af3ef8661ba4a62ab Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Thu, 22 Dec 2022 14:34:01 +0700
Subject: [PATCH 2/2] Fix tiffprefix ability

---
 libtiff/tif_hash_set.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libtiff/tif_hash_set.h b/libtiff/tif_hash_set.h
index f60e2c67..45b81744 100644
--- a/libtiff/tif_hash_set.h
+++ b/libtiff/tif_hash_set.h
@@ -31,6 +31,8 @@
 #ifndef TIFF_HASH_SET_H_INCLUDED
 #define TIFF_HASH_SET_H_INCLUDED
 
+#include "tiff.h"
+
 #include <stdbool.h>
 
 /**
-- 
2.42.0

