From 3bea1fabcf84decb5fced3215849dfa4cae35378 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Thu, 22 Dec 2022 14:34:01 +0700
Subject: [PATCH 2/2] Fix tiffprefix ability

---
 libtiff/tif_hash_set.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libtiff/tif_hash_set.h b/libtiff/tif_hash_set.h
index 1a3f8da6..4068fa82 100644
--- a/libtiff/tif_hash_set.h
+++ b/libtiff/tif_hash_set.h
@@ -31,6 +31,8 @@
 #ifndef TIFF_HASH_SET_H_INCLUDED
 #define TIFF_HASH_SET_H_INCLUDED
 
+#include "tiff.h"
+
 #include <stdbool.h>
 
 /**
-- 
2.30.2

