diff -Nur LibRaw-0.19.0_orig/src/libraw_cxx.cpp LibRaw-0.19.0/src/libraw_cxx.cpp
--- LibRaw-0.19.0_orig/src/libraw_cxx.cpp	2018-07-15 00:43:22.000000000 +0700
+++ LibRaw-0.19.0/src/libraw_cxx.cpp	2018-07-15 00:50:54.000000000 +0700
@@ -3275,7 +3275,7 @@
     {
       if (!imgdata.rawdata.ph1_cblack || !imgdata.rawdata.ph1_rblack)
       {
-        register int bl = imgdata.color.phase_one_data.t_black;
+        int bl = imgdata.color.phase_one_data.t_black;
         for (int row = 0; row < S.raw_height; row++)
         {
           checkCancel();
@@ -3289,7 +3289,7 @@
       }
       else
       {
-        register int bl = imgdata.color.phase_one_data.t_black;
+        int bl = imgdata.color.phase_one_data.t_black;
         for (int row = 0; row < S.raw_height; row++)
         {
           checkCancel();
@@ -6115,7 +6115,7 @@
 
 static void *lr_memmem(const void *l, size_t l_len, const void *s, size_t s_len)
 {
-  register char *cur, *last;
+  char *cur, *last;
   const char *cl = (const char *)l;
   const char *cs = (const char *)s;
 
