diff -Nur lcms2-2.11_orig/include/lcms2.h lcms2-2.11/include/lcms2.h
--- lcms2-2.11_orig/include/lcms2.h	2020-06-17 00:10:37.000000000 +0700
+++ lcms2-2.11/include/lcms2.h	2020-06-24 19:11:48.055489874 +0700
@@ -62,7 +62,7 @@
 // #define CMS_RELY_ON_WINDOWS_STATIC_MUTEX_INIT
 
 // Uncomment this to remove the "register" storage class
-// #define CMS_NO_REGISTER_KEYWORD 1
+#define CMS_NO_REGISTER_KEYWORD 1
 
 // ********** End of configuration toggles ******************************
 
