diff -Nur giflib-5.1.4_orig/lib/gifalloc.c giflib-5.1.4/lib/gifalloc.c
--- giflib-5.1.4_orig/lib/gifalloc.c	2018-06-17 21:24:41.341134000 +0700
+++ giflib-5.1.4/lib/gifalloc.c	2018-06-17 21:39:32.560770696 +0700
@@ -9,6 +9,7 @@
 #include <string.h>
 
 #include "gif_lib.h"
+#include "gif_lib_private.h"
 
 #define MAX(x, y)    (((x) > (y)) ? (x) : (y))
 
diff -Nur giflib-5.1.4_orig/lib/gif_lib.h giflib-5.1.4/lib/gif_lib.h
--- giflib-5.1.4_orig/lib/gif_lib.h	2018-06-17 21:24:41.341134000 +0700
+++ giflib-5.1.4/lib/gif_lib.h	2018-06-17 21:38:25.358403551 +0700
@@ -244,9 +244,6 @@
                                      GifPixelType ColorTransIn2[]);
 extern int GifBitSize(int n);
 
-extern void *
-reallocarray(void *optr, size_t nmemb, size_t size);
-
 /******************************************************************************
  Support for the in-core structures allocation (slurp mode).              
 ******************************************************************************/
diff -Nur giflib-5.1.4_orig/lib/gif_lib_private.h giflib-5.1.4/lib/gif_lib_private.h
--- giflib-5.1.4_orig/lib/gif_lib_private.h	2018-06-17 21:24:41.341134000 +0700
+++ giflib-5.1.4/lib/gif_lib_private.h	2018-06-17 21:38:56.803124760 +0700
@@ -54,6 +54,9 @@
     bool gif89;
 } GifFilePrivateType;
 
+extern void *openbsd_reallocarray(void *optr, size_t nmemb, size_t size);
+#define reallocarray openbsd_reallocarray
+
 #endif /* _GIF_LIB_PRIVATE_H */
 
 /* end */
diff -Nur giflib-5.1.4_orig/lib/openbsd-reallocarray.c giflib-5.1.4/lib/openbsd-reallocarray.c
--- giflib-5.1.4_orig/lib/openbsd-reallocarray.c	2018-06-17 21:24:41.341134000 +0700
+++ giflib-5.1.4/lib/openbsd-reallocarray.c	2018-06-17 21:39:58.417921999 +0700
@@ -27,7 +27,7 @@
 #define MUL_NO_OVERFLOW	((size_t)1 << (sizeof(size_t) * 4))
 
 void *
-reallocarray(void *optr, size_t nmemb, size_t size)
+openbsd_reallocarray(void *optr, size_t nmemb, size_t size)
 {
 	if ((nmemb >= MUL_NO_OVERFLOW || size >= MUL_NO_OVERFLOW) &&
 	    nmemb > 0 && SIZE_MAX / nmemb < size) {
