From 95d9d277c6a2fe2af70e25876942d2b413dacbaa Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Sun, 14 May 2017 23:13:29 +0700
Subject: [PATCH 2/5] Use custom third-party libraries paths

---
 src/3rdparty/zlib_dependency.pri | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/3rdparty/zlib_dependency.pri b/src/3rdparty/zlib_dependency.pri
index 2c714b3..34187a4 100644
--- a/src/3rdparty/zlib_dependency.pri
+++ b/src/3rdparty/zlib_dependency.pri
@@ -1,6 +1,7 @@
 # zlib dependency satisfied by bundled 3rd party zlib or system zlib
-qtConfig(system-zlib) {
-    QMAKE_USE_PRIVATE += zlib
-} else {
-    QT_PRIVATE += zlib-private
-}
+#qtConfig(system-zlib) {
+#    QMAKE_USE_PRIVATE += zlib
+#} else {
+#    QT_PRIVATE += zlib-private
+#}
+include($${PWD}/../../../../zlib/zlib.pri)
-- 
2.19.2

