From 394a003de5c0288e9edc72afe0f7b26d7c333c2e Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Wed, 24 Nov 2021 02:21:35 +0700
Subject: [PATCH 1/2] Fix build with MSVC 2019

---
 src/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/config.h b/src/config.h
index 22478f1..85e0380 100644
--- a/src/config.h
+++ b/src/config.h
@@ -40,7 +40,7 @@
 #define LARGE_BINARY 1
 
 #ifndef __clang__
-#define USE_SIMD 1
+//#define USE_SIMD 1
 #endif
 
 /**************************/
-- 
2.34.0

