From 77e945e285c7eee1ea7f0054c06269a22fa726b9 Mon Sep 17 00:00:00 2001
From: Peter Zhigalov <peter.zhigalov@gmail.com>
Date: Wed, 11 May 2022 01:33:59 +0700
Subject: [PATCH 3/3] Fix build with TDM GCC 5.1.0

---
 src/image/crc32k.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/image/crc32k.hpp b/src/image/crc32k.hpp
index 65e6fee..2ffe1a7 100644
--- a/src/image/crc32k.hpp
+++ b/src/image/crc32k.hpp
@@ -4,10 +4,15 @@
 
 // define endianess and some integer data types
 #if defined(_MSC_VER) || defined(__MINGW32__)
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
   typedef unsigned __int8  uint8_t;
   typedef unsigned __int16 uint16_t;
   typedef unsigned __int32 uint32_t;
   typedef   signed __int32  int32_t;
+#else
+  // uint8_t, uint32_t, in32_t
+  #include <stdint.h>
+#endif
 
   #define __ORDER_BIG_ENDIAN__ 4321
   #define __ORDER_LITTLE_ENDIAN__ 1234
-- 
2.36.0

